httpx/concurrency/asyncio.py

```diff
@@ -170,6 +171,8 @@ def is_connection_dropped(self) -> bool:
 
     async def close(self) -> None:
         self.stream_writer.close()
+        if sys.version_info >= (3, 7):
+            await self.stream_writer.wait_closed()
```

Comment:
So, is waiting for the write half of the connection to close good enough? Or could we still get the `K5S_INIT` error code somehow, as you mentioned in https://github.com/encode/httpx/pull/369#discussion_r327187795?

Reply:
Or is this part of the "try and see" note at the end of the PR description? :)

Reply:
Try and see 😛 but I'm really in two minds about it 😕. We could still get the error here if we somehow close the stream before receiving all the data from the other side.

I _think_ it should be fine to catch the error here. Once the error happens, the error will persist in the connection and whenever we try to read/write to the stream it will be raised again. But this seems correct since we shouldn't try to use a closed stream.

Reply:
OK. I'm good with merging in this state, given that I think we have enough context to come back here and try something else out if users start encountering issues.